packages/graphql-server/src/plugins/__tests__/useArmor.test.ts

```diff
@@ -0,0 +1,671 @@
+                title: 'Ba',
+                body: 'Zinga',
+                author: { id: 1, name: 'Bob' },
+              }
+            },
+          },
+        },
+      }),
+  }
+})
+
+jest.mock('../../directives/makeDirectives', () => {
+  return {
+    makeDirectivesForPlugin: () => [],
+  }
+})
+
+interface MockLambdaParams {
+  headers?: { [key: string]: string }
+  body?: string | null
+  httpMethod: string
+  [key: string]: any
+}
+
+const mockLambdaEvent = ({
+  headers,
+  body = null,
+  httpMethod,
+  ...others
+}: MockLambdaParams): APIGatewayProxyEvent => {
+  return {
+    headers: headers || {},
+    body,
+    httpMethod,
+    multiValueQueryStringParameters: null,
+    isBase64Encoded: false,
+    multiValueHeaders: {}, // this is silly - the types require a value. It definitely can be undefined, e.g. on Vercel.
+    path: '/graphql',
+    pathParameters: null,
+    stageVariables: null,
+    queryStringParameters: null,
+    requestContext: null as any,
+    resource: null as any,
+    ...others,
+  }
+}
+
+describe('useArmor secures the GraphQLHandler endpoint for depth, aliases, cost, and other optimistic protections', () => {
+  const createHandler = (armorConfig) => {
+    return createGraphQLHandler({
+      armorConfig,
+      loggerConfig: {
+        logger: createLogger({}),
+        options: {},
+      },
+      sdls: {},
+      directives: {},
+      services: {},
+      onException: () => {},
+    })
+  }
+
+  const mockedEvent = mockLambdaEvent({
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    httpMethod: 'POST',
+  })
+
+  const mockQuery = (query) => {
+    const event = mockedEvent
+    mockedEvent.body = JSON.stringify({ query })
+    return event
+  }
+
+  const mockGraphQLRequest = async (query, armorConfig?) => {
+    const handler = createHandler(armorConfig)
+    return await handler(mockQuery(query), {} as Context)
+  }
+
+  describe('when blocking field suggestion', () => {
+    describe('with defaults', () => {
+      it('returns the field suggestion masked', async () => {
+        const query = '{ posts { id, toootle } }'
+        const response = await mockGraphQLRequest(query)
+        const { data, errors } = JSON.parse(response.body)
+
+        expect(response.statusCode).toBe(200)
+        expect(data).toBeUndefined()
+        expect(errors[0].message).toEqual(
+          'Cannot query field "toootle" on type "Post". [Suggestion hidden]?'
+        )
+      })
+    })
+
+    describe('when disabled', () => {
+      it('returns the field suggestion', async () => {
+        const query = '{ posts { id, toootle } }'
+        const armorConfig = {
+          blockFieldSuggestion: { enabled: false },
+        }
+        const response = await mockGraphQLRequest(query, armorConfig)
+        const { data, errors } = JSON.parse(response.body)
+
+        expect(response.statusCode).toBe(200)
+        expect(data).toBeUndefined()
+        expect(errors[0].message).toEqual(
```

Comment:
@dthyresson awesome tests!

Wondering for the messages, is it better to use a `matchInlineSnapshot`? That way if in future versions they changed the message, its just a `yarn test -u` to update the tests?

Reply:
Good idea. Done!